client/js/components/GameView/MoveChoiceOverlay.vue

```diff
@@ -70,181 +70,163 @@ export default {
-      const scuttleDisabled = !this.isPlayersTurn || !this.hasValidScuttleTarget;
+    },
+    scuttleMove() {
+      const scuttleDisabled = this.allMovesAreDisabled || !this.hasValidScuttleTarget;
       let scuttleDisabledExplanation = 'You can only scuttle smaller point cards';
       if (this.$store.getters.opponent.points.length === 0) {
         scuttleDisabledExplanation = 'Your opponent has no point cards to scuttle';
       }
-      if (allMovesAreDisabled) {
-        scuttleDisabledExplanation = allMovesDisabledExplanation;
+      if (this.allMovesAreDisabled) {
+        scuttleDisabledExplanation = this.disabledText;
       }
-      const scuttleMove = {
+      return {
         displayName: 'Scuttle',
         eventName: 'scuttle',
         moveDescription: 'Scrap a lower point card',
-        disabled: scuttleDisabled || allMovesAreDisabled,
+        disabled: scuttleDisabled,
         disabledExplanation: scuttleDisabledExplanation,
       };
+    },
+    oneOffMove() {
+      return {
+        displayName: 'One-Off',
+        eventName: 'oneOff',
+        moveDescription: this.selectedCard.ruleText,
+        disabled: this.allMovesAreDisabled,
+        disabledExplanation: this.disabledText,
+      };
+    },
+
+    targetedOneOffMove() {
+      let oneOffDisabled = this.allMovesAreDisabled;
+      let oneOffDisabledExplanation = this.disabledText;
+      if (!this.allMovesAreDisabled) {
+        if (this.opponentQueenCount >= 2) {
+          oneOffDisabled = true;
+          oneOffDisabledExplanation = `You can't play a ${this.selectedCard.rank} while your opponent has two or more queens`;
+        } else {
+          let validTargetExists;
+          // Twos
+          if (this.selectedCard.rank === 2) {
+            const numOpFaceCards = this.$store.getters.opponent.faceCards.length;
+            const numOpJacks = this.$store.getters.opponent.points.reduce(
+              (jackCount, pointCard) => {
+                return jackCount + pointCard.attachments.length;
+              },
+              0
+            );
+            const numTotalTargets = numOpFaceCards + numOpJacks;
+            validTargetExists = numTotalTargets >= 1;
+            if (!validTargetExists) {
+              oneOffDisabled = true;
+              oneOffDisabledExplanation = 'There are no Royals to target';
+            }
+            // Nines
+          } else {
+            const numValidTargets =
+              this.$store.getters.opponent.points.length +
+              this.$store.getters.opponent.faceCards.length;
+            if (numValidTargets === 0) {
+              oneOffDisabled = true;
+              oneOffDisabledExplanation = 'There are no point cards or Royals to target';
+            }
+          }
+        }
+      }
+      return {
+        displayName: 'One-Off',
+        eventName: 'targetedOneOff',
+        moveDescription: this.selectedCard.ruleText,
+        disabled: oneOffDisabled || this.allMovesAreDisabled,
```

Comment:
Because you assign `oneOffDisabled` in the beginning, and only re-assign it if you're setting it to true, you can just
```suggestion
        disabled: oneOffDisabled,
```